src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_IInterpolatedStringExpression.cs

```diff
@@ -444,5 +444,635 @@ public void M(int x)
+              Left: 
+                IFlowCaptureReferenceOperation: 0 (OperationKind.FlowCaptureReference, Type: System.String, IsImplicit) (Syntax: 'a')
+              Right: 
+                IInterpolatedStringOperation (OperationKind.InterpolatedString, Type: System.String) (Syntax: '$""{M2($""{b ?? c}"")}""')
+                  Parts(1):
+                      IInterpolationOperation (OperationKind.Interpolation, Type: null) (Syntax: '{M2($""{b ?? c}"")}')
+                        Expression: 
+                          IInvocationOperation ( System.Int32 Class.M2(System.String z)) (OperationKind.Invocation, Type: System.Int32) (Syntax: 'M2($""{b ?? c}"")')
+                            Instance Receiver: 
+                              IFlowCaptureReferenceOperation: 1 (OperationKind.FlowCaptureReference, Type: Class, IsImplicit) (Syntax: 'M2')
+                            Arguments(1):
+                                IArgumentOperation (ArgumentKind.Explicit, Matching Parameter: z) (OperationKind.Argument, Type: null) (Syntax: '$""{b ?? c}""')
+                                  IInterpolatedStringOperation (OperationKind.InterpolatedString, Type: System.String) (Syntax: '$""{b ?? c}""')
+                                    Parts(1):
+                                        IInterpolationOperation (OperationKind.Interpolation, Type: null) (Syntax: '{b ?? c}')
+                                          Expression: 
+                                            IFlowCaptureReferenceOperation: 3 (OperationKind.FlowCaptureReference, Type: System.Int32, IsImplicit) (Syntax: 'b ?? c')
+                                          Alignment: 
+                                            null
+                                          FormatString: 
+                                            null
+                                  InConversion: CommonConversion (Exists: True, IsIdentity: True, IsNumeric: False, IsReference: False, IsUserDefined: False) (MethodSymbol: null)
+                                  OutConversion: CommonConversion (Exists: True, IsIdentity: True, IsNumeric: False, IsReference: False, IsUserDefined: False) (MethodSymbol: null)
+                        Alignment: 
+                          null
+                        FormatString: 
+                          null
+
+    Next (Regular) Block[B5]
+Block[B5] - Exit
+    Predecessors: [B4]
+    Statements (0)
+";
+            var expectedDiagnostics = DiagnosticDescription.None;
+
+            VerifyFlowGraphAndDiagnosticsForTest<BlockSyntax>(source, expectedFlowGraph, expectedDiagnostics);
+        }
+
+        [CompilerTrait(CompilerFeature.IOperation, CompilerFeature.Dataflow)]
+        [Fact]
+        public void InterpolatedStringExpression_InvalidExpressionInInterpolation_Flow()
+        {
+            string source = @"
+using System;
+
+internal class Class
+{
+    public void M(string x)
+    /*<bind>*/{
+        x = $""{x1 ?? x}"";
```

Comment:
>x1 ?? x [](start = 16, length = 7)

Is there anything interesting in using coalesce vs. conditional expression that we already covered.

Reply:
This unit test is covering conditional code within an interpolation with error `ErrorCode.ERR_NameNotInContext`. This is not covered by any other unit test.

---
In reply to: [185918540](https://github.com/dotnet/roslyn/pull/26545#discussion_r185918540) [](ancestors = 185918540)

Reply:
> This unit test is covering conditional code within an interpolation with error ErrorCode.ERR_NameNotInContext. This is not covered by any other unit test.

Why is this interesting? Is there some kind of unique interaction going on?

---
In reply to: [185919234](https://github.com/dotnet/roslyn/pull/26545#discussion_r185919234) [](ancestors = 185919234,185918540)

Reply:
To clarify, when I said: "over control flow in each part where it can go, even in error cases." I meant cover even cases when that control flow is going to be an error in that position. If it is valid to have a control flow in a position, there is no need to test with an error inside it

---
In reply to: [185920295](https://github.com/dotnet/roslyn/pull/26545#discussion_r185920295) [](ancestors = 185920295,185919234,185918540)

Reply:
> no need to test with an error inside it

I felt it would be good to have a scenario with an `IInvalidOperation` within an `IInterpolationOperation`. I will remove this test as per your suggestion.

---
In reply to: [185921234](https://github.com/dotnet/roslyn/pull/26545#discussion_r185921234) [](ancestors = 185921234,185920295,185919234,185918540)